### ISSUE TITLE
`Array` vs `Float32Array`: Performance measurements (again)

### ISSUE THREAD

**Issue by psnet** [toji/gl-matrix#465]
TL;DR
`Float32Array` is faster for IO operations than usual `Array`. Only creation of new objects is slower for `Float32Array`.

Measurements
---

I saw issues there about complains of performance and I did recent clean and exact measurements about casual `Array` vs `Float32Array`:

https://www.measurethat.net/Benchmarks/Show/30129/3/array-vs-float32array-performance-test

And IO operations with `Float32Array` are `2x` **FASTER** than usual `Array`. `Array` wins **only** in **creation of new objects** (arrays). But if we create needed data (preallocated) before main render loop (cycle) then `Float32Array` is big win.

So I want to say author should not remove support of `Float32Array` and still use approach available now: default is `Float32Array` but you can change it by method `glMatrix.setMatrixArrayType(Array)`

**Comment by snowfrogdev** [comment]
> ## TL;DR
> `Float32Array` is faster for IO operations than usual `Array`. Only creation of new objects is slower for `Float32Array`.
> 
> ## Measurements
> I saw issues there about complains of performance and I did recent clean and exact measurements about casual `Array` vs `Float32Array`:
> 
> https://www.measurethat.net/Benchmarks/Show/30129/3/array-vs-float32array-performance-test
> 
> And IO operations with `Float32Array` are `2x` **FASTER** than usual `Array`. `Array` wins **only** in **creation of new objects** (arrays). But if we create needed data (preallocated) before main render loop (cycle) then `Float32Array` is big win.
> 
> ## Summary
> So I want to say author should not remove support of `Float32Array` and still use approach available now: default is `Float32Array` but you can change it by method `glMatrix.setMatrixArrayType(Array)`

When I run your benchmark suite on my computer (Windows - Chrome 129) the `Array` beats the `TypedArray` in all tests except for sort.

![image](https://github.com/user-attachments/assets/933e9c7c-7f55-4348-9868-5f504e163864)
